azext_iot/digitaltwins/providers/resource.py

```diff
@@ -71,6 +72,25 @@ def create(
                 long_running_operation_timeout=timeout,
             )
 
+            def check_state(lro):
+                from time import sleep
+                instance = lro.resource().as_dict()
+                state = instance.get('provisioning_state', None)
+                retries = 0
+                while state == "Provisioning" and retries < MAX_ADT_CREATE_RETRIES:
```

Comment:
Think it would be good to have the provisioning state values we care about in an enum